screens/locationData.js

```diff
@@ -0,0 +1,179 @@
+import React, { PureComponent } from 'react';
+import { View, Text, TouchableOpacity, StyleSheet, Alert } from 'react-native';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import I18n from '../assets/i18n/i18n';
+import Icon from '../components/Icon';
+import colours from '../styles/colours';
+import fonts from '../styles/fonts';
+import { UploadStatus } from '../actions/uploads';
+import { pushLocalLocations } from '../actions/locations';
+import { requestPushPermission } from '../actions/permissions';
+import CollapsibleList from '../components/CollapsibleList';
+
+const styles = StyleSheet.create({
+  container: {
+    width: '100%',
+    height: '100%',
+    flexDirection: 'column',
+  },
+  button: {
+    alignSelf: 'flex-start',
+    borderWidth: 1,
+    borderRadius: 5,
+    borderColor: colours.black,
+    backgroundColor: colours.greys.lightest,
+    padding: 10,
+    marginHorizontal: 5,
+    marginVertical: 10,
+  },
+  first: {
+    borderTopWidth: 1,
+  },
+  section: {
+    flexDirection: 'row',
+    paddingVertical: 15,
+    paddingHorizontal: 10,
+    backgroundColor: colours.greys.lightest,
+    borderBottomWidth: 1,
+    borderColor: colours.black,
+    alignItems: 'center',
+  },
+  sectionText: {
+    fontSize: fonts.large,
+  },
+});
+
+class LocationData extends PureComponent {
+  state = {
+    failedOpen: false,
+    numberUploading: 0,
+  }
+
+  onFailedPressed = () => {
+    this.setState(p => ({ failedOpen: !p.failedOpen }));
+  }
+
+  uploadLocations = () => {
+    this.setState({ numberUploading: this.props.stats.offline + this.props.stats.failed });
+    this.props.showPushDialog();
+  }
+
+  render() {
+    const {
+      failedLocations,
+      stats,
+      uploading,
+    } = this.props;
+    const {
+      failedOpen,
```

Comment:
Why not just let it scroll instead of opening and closing? I didn't think mobile apps normally did that.

Reply:
I figured it might be nice to have it hidden initially if for some reason there is a lot of failed locations. That way the screen will load faster and if they aren't there to look at failed items they don't have to. IDK should I get rid of it?

Reply:
I'd get rid of it for now. The `FlatList` has some great features to making loading the data smoother.

Reply:
Just plain FlatList now